src/executables/sysrepocfg.c

```diff
@@ -709,15 +710,15 @@ arg_is_file(const char *optarg)
 static int
 arg_get_ds(const char *optarg, sr_datastore_t *ds)
 {
-    if (!strcmp(optarg, "running")) {
+    if (optarg[0] == 'r') {
         *ds = SR_DS_RUNNING;
-    } else if (!strcmp(optarg, "startup")) {
+    } else if (optarg[0] == 's') {
         *ds = SR_DS_STARTUP;
-    } else if (!strcmp(optarg, "candidate")) {
+    } else if (optarg[0] == 'c') {
         *ds = SR_DS_CANDIDATE;
-    } else if (!strcmp(optarg, "operational")) {
+    } else if (optarg[0] == 'o') {
         *ds = SR_DS_OPERATIONAL;
-    } else if (!strcmp(optarg, "factory-default")) {
+    } else if (optarg[0] == 'f') {
```

Comment:
Please restrict the valid values to `running` or `r`, similarly for all the datastores. Accepting any mangled words starting withe the right letters could cause problems in future.

Reply:
Makes sense, updated